smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpRpcProtocolGenerator.java

```diff
@@ -404,43 +417,106 @@ private void generateErrorDeserializer(GenerationContext context, StructureShape
+                deserializationDocumentShapes.add(outputShape);
+
+                if (useEventStreamDeserializer) {
+                    writer.write("const contents: any = {};");
+                    Collection<MemberShape> memberValues = outputShape.getAllMembers().values();
+                    Optional<MemberShape> first = memberValues.stream().findFirst();
+                    String memberName = first.get().getMemberName();
+
+                    generateRpcEventStreamDeserializer(
+                        context,
+                        outputShape,
+                        memberName,
+                        "contents." + memberName
+                    );
+                } else {
                     writer.write("const data: any = await parseBody(output.body, context)");
                     writer.write("let contents: any = {};");
-
-                    // If there's an output present, we know it's a structure.
-                    StructureShape outputShape = context.getModel().expectShape(outputId).asStructureShape().get();
-
-                    // Track output shapes so their deserializers may be generated.
-                    deserializingDocumentShapes.add(outputShape);
-
                     deserializeOutputDocument(context, operation, outputShape);
-                },
-                () -> {
-                    // If there is no output, the body still needs to be collected so the process can exit.
-                    writer.write("await collectBody(output.body, context);");
-                });
+                }
+            },
+            () -> {
+                // If there is no output, the body still needs to be collected so the process can exit.
+                writer.write("await collectBody(output.body, context);");
+            });
+    }
+
+    private void generateRpcEventStreamDeserializer(GenerationContext context,
+                                                    StructureShape event,
+                                                    String memberName,
+                                                    String assignTo) {
+        TypeScriptWriter writer = context.getWriter();
+        Symbol targetSymbol = context.getSymbolProvider().toSymbol(event);
+
+        writer.openBlock(assignTo + " = context.eventStreamMarshaller.deserialize(", ");", () -> {
+            writer.write("output.body,");
+            writer.openBlock("async event => {", "}", () -> {
+                writer.addImport("toEventMessage", "__toEventMessage", "@aws-sdk/smithy-client");
+
+                writer.write("const parsedEvent = __toEventMessage(event);");
+                String deserFunctionName = ProtocolGenerator.getDeserFunctionName(
+                    targetSymbol, context.getProtocolName()
+                );
+                writer.write("return $1L({ $2L: parsedEvent }, context).$2L;",
+                    deserFunctionName, memberName
```

Comment:
Can you refer to the http binding protocol to handle possible single-event stream?: https://github.com/awslabs/smithy-typescript/blob/acf2ede7fa3b2d833c255840f2f3e3ecb221f63e/smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java#L2473-L2483

For the multi-event stream, the target shape is a union. The `parsedEvent` should look like `{ [eventName]: Message }`, whereas the single-event stream's target shape is a structure, the `parsedEvent` here should be a single `Message` interface(so it's better to rename to `eventMessage`).

Reply:
note: we clarified with the Smithy team that single event stream is no longer a scenario we need to cover